### PR TITLE
Ruby on Rails Capstone Project: Budget App

### PR DESCRIPTION
**Pull request by mikemtzp**
# :memo: Project Description

> The budget app is a mobile web application where you can manage your budget: you have a list of transactions associated with a category so that you can see how much money you spent and on what.

This app allows the user to:
- register and log in, so that the data is private to them.
- introduce new transactions associated with a category.
- see the money spent on each category.

### Design
- Follow these [design guidelines](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=) by [Gregoire Vella on Behance](https://www.behance.net/gregoirevella) including:
  - Colors.
  - Typography: font face, size and weight.
  - Layout: composition and space between elements.

### Interactions

- Splash screen
  - A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.
- Sign up and log in pages
  - The user should be able to register in the app with full name, email and password (all mandatory).
  - The user can log into the app using email and password.
  - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).
 
- Home page (catagories page)
  - When the user logs in, they are presented with the categories page.
  - For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
  - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
  - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- Transactions page
  - For a given category, the list of transactions is presented, ordered by the most recent.
  - At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
  - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new category" page
  - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
  - The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new transaction" page
  - The user fills out a form to create a new transaction with:
    - name (mandatory)
    - amount (mandatory)
    - categories (mandatory at least one)
  - The user click (or taps) the "Save" button to create the new transation, and is taken to the transactions page for that category.
  - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

### Testing requirements
- Create unit and integration tests for all the most important components.

### Technical requirements

- You should use Postgres as your database.
- You should use Devise for authentication.
- You should validate all user input to make sure that anyone with bad intentions cannot compromise your app.
- You can use a view template engine of your choice (.erb, .slim, .haml).
- The project should be deployed and accessible online.
- Your database schema should reflect the following structure:

### CODE REVIEW

File: app/views/expenses/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  <% else %>
+    <% @expenses.each do |expense| %>
+      <%= render expense %>
+    <% end %>
+  <% end %>
+</div>
+
+<div class="add-category">
+  <%= link_to "Add transaction", new_group_expense_path, class: "splash-login add-category-button"%>
+</div>
```
Comment (AnselemOdims) [inline app/views/expenses/index.html.erb:13]: - [x] Good job adding the required functionalities, but please ensure that your design stick to the original guidelines. 
- [x] The button `Add transaction` should be at the bottom of the page.
- [x] The icons should also show

### Your Design
image

### Original Design
image

File: app/views/groups/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    <p class="center-message"><em>No groups available for the moment</em></p>
+  <% else %>
+    <% @groups.each do |group| %>
+      <%= render group, group: group %>
+    <% end %>
+  <% end %>
+</div>
+
+<div class="add-category">
+  <%= link_to "Add category", new_group_path, class: "splash-login add-category-button" %>
```
Comment (AnselemOdims) [inline app/views/groups/index.html.erb:12]: - [x] As pointed out above, please ensure that the `Add category` button sticks to the bottom of the page.

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @mikemtzp , 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [x] Please ensure that the live deployment link is included in the README file
- [x] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: app/views/expenses/index.html.erb (hunk 1 above)
Comment (mikemtzp) [reply app/views/expenses/index.html.erb:13]: Hi, @AnselemOdims!
Thanks for the observation. Now, the icons are shown, and I fixed the position of the button to the bottom. 😁

File: app/views/groups/index.html.erb (hunk 2 above)
Comment (mikemtzp) [reply app/views/groups/index.html.erb:12]: @AnselemOdims Also fixed the add category button to be at the bottom. 👍🏼

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @mikemtzp ,
# STATUS: APPROVED! 🥇 💯 
done

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**